pkg/server/server.go

```diff
@@ -911,12 +938,32 @@ func updateDeployment(w http.ResponseWriter, r *http.Request) {
 			//Get JSON from url
 			httpClient := &http.Client{}
 
-			req, err := http.NewRequest("GET", tempJSON.PtsURL, nil)
+			ptsURL, err := url.Parse(tempJSON.PtsURL)
+			if err != nil {
+				errorMessage := fmt.Sprintf("Error parsing ptsURL\n")
+				http.Error(w, errorMessage, http.StatusInternalServerError)
+				helper.LogError.Printf(errorMessage)
+				return
+			}
+
+			internalRouterFlag := false
+
+			if ptsURL.Host == shipyardHost {
+				ptsURL.Host = internalRouterHost
+				ptsURL.Scheme = "http"
+				internalRouterFlag = true
+			}
+
+			req, err := http.NewRequest("GET", ptsURL.String(), nil)
+
+			if internalRouterFlag {
+				req.Host = shipyardHost
+				req.Header.Add("Host", shipyardHost)
+				req.Header.Add("X-ROUTING-API-KEY", base64.StdEncoding.EncodeToString([]byte(shipyardPrivateSecret)))
+			}
```

Comment:
This code and the code above it seems to be the same.  Would it make sense to refactor it into something reusable?